realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1344,6 +1248,13 @@ public void run() {
                                 });
                             }
                         }
+
+                        if (!handlerController.threadContainsAsyncQueries() && !asyncTransactionCallbacks.isEmpty()) {
+                            for (Runnable transactionCallback : asyncTransactionCallbacks) {
+                                handler.post(transactionCallback);
```

Comment:
`HandlerController.realmChanged()` is invoked by `BaseRealm.commitTransaction` and `BaseRealm.refresh`. If we move this logic to `HandlerController.realmChanged()` and there is no other transactions and refreshes, callbacks won't be invoked.

I made `HandlerController.executeAsyncTransactionCallbacks` for `HandlerController.completedAsyncQueriesUpdate` and `executeTransactionAsync` instead. How about this?